### DISCUSSION TITLE
Support statement regarding PHP 7.4 EOL

### DISCUSSION THREAD

**Discussion by amartinr** [oerdnj/deb.sury.org#1784] [Q&A] [answered]
Hi,

Just wanted to know what would be your's and Debian's policy (if possible) on supporting PHP 7.4 after upstream EOL is reached. We're currently planning to upgrade to 8.x and wanted to have an idea of the potential deadlines we might be facing in the near future.

After having read  [https://github.com/oerdnj/deb.sury.org/discussions/1740](https://github.com/oerdnj/deb.sury.org/discussions/1740) and [https://launchpad.net/~ondrej/+archive/ubuntu/php](https://launchpad.net/~ondrej/+archive/ubuntu/php) I understand that you won't be providing support after upstream's EOL date is reached. Is that correct?

To make my question clearer, I'm referring to PHP 7.4 available at [http://packages.sury.org/php/](http://packages.sury.org/php/) for bullseye.

Thanks!

**Comment by oerdnj** [comment, marked as answer]
* DEB.SURY.ORG policy is to support PHP versions as long as it is feasible, but I am not going to support **distributions** that are past end-of-life.

* Debian (as pristine vanilla Debian) policy is to support the software that's in the archive for the lifetime of the release, e.g. PHP 7.4 will be supported for the lifetime of Debian bullseye

If you need support that has defined SLAs and guarantees, there's [PHP LTS by Freexian](https://php.freexian.com); the current guarantee is that any PHP release will be supported at least for two extra years past the official upstream end-of-life.

**Reply by amartinr** [reply]
Crystal clear. Many thanks for your prompt answer!

**Comment by sleemanj** [comment]
Support for distro's ends generally when the distro goes EOL, but PHP versions tend to be something barely short of eternal.  

5.6 has been EOL for how many years now, but @oerdnj very capably and generously continues to support it in his repo and even is known to roll patches into it.

**Reply by amartinr** [reply]
Thanks! I know severe security bugs might get backported after EOL if the security team think the risk posed by threats justifies the effort.
